presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -669,4 +671,62 @@ public static Boolean castTokenAsBoolean(JsonParser parser)
                 throw new PrestoException(INVALID_CAST_ARGUMENT, format("Unexpected token when cast to %s: %s", StandardTypes.BOOLEAN, parser.getText()));
         }
     }
+
+    public static Long castTokenAsShortDecimal(JsonParser parser, int precision, int scale)
+            throws IOException
+    {
+        BigDecimal bigDecimal = castTokenAsJavaDecimal(parser, precision, scale);
+        if (bigDecimal != null && bigDecimal.precision() > precision) {
+            // TODO: Should we use NUMERIC_VALUE_OUT_OF_RANGE instead?
+            throw new PrestoException(INVALID_CAST_ARGUMENT, format("Cannot cast input json to DECIMAL(%s,%s)", precision, scale));
```

Comment:
@losipiuk Agree. Throwing `NumberFormatException` sounds a good way. 

We need to also consider what kind of exception should be thrown for `CastTokenAsXXX`